### PR TITLE
Update ipfs-lite, test data races, and other implications

### PR DESCRIPTION
**Pull request by jsign**
- Update `ipfs-lite` which cascaded to a lot of other important updates
- Fix all broken things from that update

### CODE REVIEW

File: common/common.go [hunk 1]
```diff
@@ -62,20 +60,14 @@ func DefaultNetwork(repoPath string, opts ...NetOption) (NetBoostrapper, error)
...
 
 	ctx, cancel := context.WithCancel(context.Background())
-	pstore, err := pstoreds.NewPeerstore(ctx, litestore, pstoreds.DefaultOpts())
-	if err != nil {
-		litestore.Close()
-		cancel()
-		return nil, err
-	}
 	priv := util.LoadKey(filepath.Join(ipfsLitePath, "key"))
 	h, d, err := ipfslite.SetupLibp2p(
```
Comment (jsign) [inline common/common.go:64]: IPFS-lite now doesn't need an explicit peerstore, but requires a batching datastore.

File: db/index.go [hunk 2]
```diff
@@ -88,7 +88,7 @@ func indexUpdate(baseKey ds.Key, path string, index Index, tx ds.Txn, key ds.Key
 		return nil
```
Comment (jsign) [inline db/index.go:88]: Indexes where relying on the _prefix query will contain non-child keys_ assumption.

File: db/query_more_test.go [hunk 3]
```diff
@@ -94,7 +94,7 @@ var (
```
Comment (jsign) [inline db/query_more_test.go:94]: Test ran in parallel, but `createCollectionWithData` was modifying the global variable!
Now that method returns the modified dataset from the sample data, so parallel runs use their own data sample.

File: db/query_test.go [hunk 4]
```diff
@@ -6,6 +6,7 @@ import (
 	"strings"
```
Comment (jsign) [inline db/query_test.go:6]: Same as file above.

File: go.mod [hunk 5]
```diff
@@ -18,35 +18,35 @@ require (
...
-	github.com/ipfs/go-log v1.0.0
-	github.com/ipfs/go-merkledag v0.2.3
-	github.com/libp2p/go-libp2p v0.4.2
-	github.com/libp2p/go-libp2p-connmgr v0.1.1
-	github.com/libp2p/go-libp2p-core v0.3.0
+	github.com/ipfs/go-ipld-cbor v0.0.4
+	github.com/ipfs/go-ipld-format v0.2.0
+	github.com/ipfs/go-log v1.0.4
+	github.com/ipfs/go-merkledag v0.3.2
+	github.com/libp2p/go-libp2p v0.8.2
```
Comment (jsign) [inline go.mod:37]: From v0.4.2 to v0.8.2

File: go.mod [hunk 6]
```diff
@@ -18,35 +18,35 @@ require (
 	github.com/gogo/status v1.1.0
 	github.com/golang/protobuf v1.3.3
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.0
-	github.com/hashicorp/go-multierror v1.0.0
-	github.com/hashicorp/golang-lru v0.5.3
-	github.com/hsanjuan/ipfs-lite v0.1.8
+	github.com/hashicorp/go-multierror v1.1.0
+	github.com/hashicorp/golang-lru v0.5.4
+	github.com/hsanjuan/ipfs-lite v1.1.11
```
Comment (jsign) [inline go.mod:23]: The innocent trigger...

File: go.mod [hunk 7]
```diff
@@ -18,35 +18,35 @@ require (
...
 	github.com/improbable-eng/grpc-web v0.12.0
 	github.com/ipfs/go-block-format v0.0.2
-	github.com/ipfs/go-blockservice v0.1.2
+	github.com/ipfs/go-blockservice v0.1.3
 	github.com/ipfs/go-cid v0.0.5
-	github.com/ipfs/go-datastore v0.3.1
-	github.com/ipfs/go-ds-badger v0.2.0
-	github.com/ipfs/go-ipfs-blockstore v0.1.1
+	github.com/ipfs/go-datastore v0.4.4
+	github.com/ipfs/go-ds-badger v0.2.4
```
Comment (jsign) [inline go.mod:29]: More deep ones. Having them in v0.x is a problem for semver...

File: logstore/lstoreds/keybook.go [hunk 8]
```diff
@@ -175,18 +175,17 @@ func (kb *dsKeyBook) ClearKeys(t thread.ID) error {
 
 // ClearLogKeys deletes all keys under a log.
 func (kb *dsKeyBook) ClearLogKeys(t thread.ID, p peer.ID) error {
-	if err := kb.clearKeys(dsLogKey(t, p, kbBase).Child(privSuffix)); err != nil {
```
Comment (jsign) [inline logstore/lstoreds/keybook.go:178]: Relied on the assumption that prefix will contain the exact key match.
Switched to simply deleting directly.

File: db/query_more_test.go (hunk 3 above)
Comment (sanderpick) [reply db/query_more_test.go:94]: 😨

File: go.mod (hunk 7 above)
Comment (sanderpick) [reply go.mod:29]: People are addicted to the idea of a v0 release

**Pull request merged**